kml.cc

```diff
@@ -1467,32 +1470,41 @@ void KmlFormat::kml_mt_simple_array(const route_head* header,
 {
   writer->writeStartElement(QStringLiteral("gx:SimpleArrayData"));
   writer->writeAttribute(QStringLiteral("name"), name);
-
   foreach (const Waypoint* wpt, header->waypoint_list) {
-
-    switch (member) {
-    case fld_power:
-      writer->writeTextElement(QStringLiteral("gx:value"), wpt->power?
-        QString::number(wpt->power, 'f', 1) : QString());
-      break;
-    case fld_cadence:
-      writer->writeTextElement(QStringLiteral("gx:value"), wpt->cadence?
-       QString::number(wpt->cadence) : QString());
-      break;
-    case fld_depth:
-      writer->writeTextElement(QStringLiteral("gx:value"), wpt->depth_has_value()?
-        QString::number(wpt->depth_value(), 'f', 1) : QString());
-      break;
-    case fld_heartrate:
-      writer->writeTextElement(QStringLiteral("gx:value"), wpt->heartrate?
-       QString::number(wpt->heartrate) : QString());
-      break;
-    case fld_temperature:
-      writer->writeTextElement(QStringLiteral("gx:value"), wpt->temperature_has_value()?
-        QString::number(wpt->temperature_value(), 'f', 1) : QString());
-      break;
-    default:
-      fatal("Bad member type");
+    const auto* fs_igc = reinterpret_cast<igc_fsdata<float>*>(wpt->fs.FsChainFind(kFsIGC));
```

Comment:
more template confusion.  hopefully we don't need igc_fsdata to be a template at all, we just always get doubles.